tools/scripts/validate_alerts.py

```diff
@@ -12,19 +14,35 @@
             "OncallTeams": { "type": "array", "items": { "type": "string" } },
             "OncallIndividuals": { "type": "array", "items": { "type": "string" } },
             "Flags": { "type": "array", "items": { "type": "string" } },
-        }
+            "branch": { "type": "string" },
+        },
+        "additionalProperties": True,
+        "required": ["AlertType", "AlertObject", "OncallTeams", "OncallIndividuals", "Flags", "branch"],
     }
 }
 
+EXTRA_ALERT_PROPERTIES = {
+    "RecurentlyFailingJobAlert" : {
+            "sha": { "type": "string" },
+        }
+    }
+
 def validate_json(json_string):
     try:
         json_object = json.loads(json_string)
         print("The input string is a valid JSON.")
     except ValueError as e:
         raise ValueError(f"The input string is not a valid JSON: Error: {e}")
 def validate_schema(json_string):
+    all_alerts_schemas = defaultdict(lambda: copy.deepcopy(BASE_ALERT_SCHEMA))
+    for alert_type, alert_schema_add_on in EXTRA_ALERT_PROPERTIES.items():
+        all_alerts_schemas[alert_type]["items"]["properties"].update(alert_schema_add_on)
+        all_alerts_schemas[alert_type]["items"]["additionalProperties"] = False
+        for property in alert_schema_add_on.keys():
```

Comment:
Nit: I guess you could extend the array here in one line

Reply:
I think this is more readable though